src/java.base/share/classes/java/io/StringWriter.java

```diff
@@ -243,4 +240,11 @@ public void flush() {
     public void close() throws IOException {
     }
 
+    private static int checkSize(int initialSize) {
+        if (initialSize < 0) {
```

Comment:
Similar checks exist e.g. in `ByteArrayOutputStream` and `CharWrite`, so could we reuse it somehow across java.io?

Reply:
I assume this is done so that "lock" is only set once during construction (StringWriter is a bit unusual in that it uses the SB as the lock object). A downside of the change is that it introduces casts. Another is that the expression to super is a bi  complicated and forces the reader to locate checkSize at the end of the file to find the code specified by the constructor. So two minds on this part, maybe it should be left unchanged.

Reply:
I have added a new proposal in the hope that the `requireNonNegative` method names and parameters should be trivial enough for users to directly understand without scrolling down. Let me know your thought on this.

Reply:
That is definitely better but I would be inclined to place `requireNonNegative()` between the constructors and the instance methods.

Reply:
It's a bit better but I think the original code is clearer/simpler, in particular it makes it clear that buf is used as the lock object. I realise this is very subjective but I think my preference would be to prefer that part and just make `buf` final.